src/main/java/io/supertokens/oauth/OAuth.java

```diff
@@ -16,295 +16,346 @@
+import io.supertokens.pluginInterface.exceptions.StorageTransactionLogicException;
+import io.supertokens.pluginInterface.jwt.JWTSigningKeyInfo;
 import io.supertokens.pluginInterface.multitenancy.AppIdentifier;
 import io.supertokens.pluginInterface.multitenancy.exceptions.TenantOrAppNotFoundException;
 import io.supertokens.pluginInterface.oauth.OAuthStorage;
 import io.supertokens.pluginInterface.oauth.exceptions.OAuth2ClientAlreadyExistsForAppException;
+import io.supertokens.session.accessToken.AccessToken;
+import io.supertokens.session.jwt.JWT;
+import io.supertokens.session.jwt.JWT.JWTException;
+import io.supertokens.signingkeys.JWTSigningKey;
+import io.supertokens.signingkeys.SigningKeys;
 import io.supertokens.utils.Utils;
 
 import java.io.IOException;
-import java.lang.reflect.InvocationTargetException;
-import java.net.URLDecoder;
-import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
 import java.security.NoSuchAlgorithmException;
-import java.security.SecureRandom;
+import java.security.spec.InvalidKeySpecException;
 import java.util.*;
+import java.util.Map.Entry;
 
 public class OAuth {
+    private static final String HYDRA_JWKS_PATH = "/.well-known/jwks.json";
+
+    private static void checkForOauthFeature(AppIdentifier appIdentifier, Main main)
+            throws StorageQueryException, TenantOrAppNotFoundException, FeatureNotEnabledException {
+        EE_FEATURES[] features = FeatureFlag.getInstance(main, appIdentifier).getEnabledFeatures();
+        for (EE_FEATURES f : features) {
+            if (f == EE_FEATURES.OAUTH) {
+                return;
+            }
+        }
 
-    private static final String LOCATION_HEADER_NAME = "Location";
-    private static final String COOKIES_HEADER_NAME = "Set-Cookie";
-    private static final String ERROR_LITERAL = "error=";
-    private static final String ERROR_DESCRIPTION_LITERAL = "error_description=";
-
-    private static final String HYDRA_AUTH_ENDPOINT = "/oauth2/auth";
-    private static final String HYDRA_CLIENTS_ENDPOINT = "/admin/clients";
-
-    public static OAuthAuthResponse getAuthorizationUrl(Main main, AppIdentifier appIdentifier, Storage storage, JsonObject paramsFromSdk)
-            throws InvalidConfigException, HttpResponseException, IOException, OAuthAuthException, StorageQueryException,
-            TenantOrAppNotFoundException {
+        throw new FeatureNotEnabledException(
+                "OAuth feature is not enabled. Please subscribe to a SuperTokens core license key to enable this " +
+                        "feature.");
+    }
 
+    public static HttpRequest.Response handleOAuthProxyGET(Main main, AppIdentifier appIdentifier, Storage storage, String path, boolean proxyToAdmin, Map<String, String> queryParams, Map<String, String> headers) throws StorageQueryException, OAuthClientNotFoundException, TenantOrAppNotFoundException, FeatureNotEnabledException, InvalidConfigException, IOException, OAuthAPIException {
```

Comment:
There seems to be a substantial overlap between this and `OAuthProxyHelper` is there a way we could de-duplicate? Not an issue if not, but we should consider it.

Reply:
Made it in such a way that ProxyHelper only deals with incoming request and response. OAuth takes care of querying the hydra.

Reply:
Maybe it's worth renaming to clarify that we are actually sending the request or querying the OAuth apis instead of handling incoming requests.

Reply:
have renamed to `doOAuthProxyGET`, `doOAuthJsonPOST`, etc.